39232 update comment regex

allen-ciq opened the pull request:
## Description

If a jsdoc comment had multiple <span> tags, all tags after the first
were not parsed.  This change updates the regex for finding span's.

Additionally, added checking for surrounding ``` to allow for literal
<span> tags in jsdocs.

Fixes # 39232

allen-ciq committed:
39232 update comment regex

If a jsdoc comment had multiple <span> tags, all tags after the first
were not parsed.  This change updates the regex for finding span's.

Additionally, added checking for surrounding ``` to allow for literal
<span> tags in jsdocs.

RMcNeely commented on src/common/common.js:
````diff
@@ -129,7 +129,8 @@ function cleanCommentData(comment, skipAdditional = []) {
 
-  comment = comment.replace(/<span[^>]*>(.*?)<\/span>/, "$1")
+  //https://regex101.com/r/aAZJcy/2
+  comment = comment.replace(/(?<!```.*)<span[^>]*>(.*)<\/span>|<span[^>]*>(.*)<\/span>(?!.*```)/g, "$1")
````
Whats' the rational for matching something that starts but not ends with a code block and ends but doesn't start with a code block?

allen-ciq replied:
This code removes `<span>` tags and replaces with the contents.  The idea here is that we want to keep the raw string if we're inside a code block.  Also, note the the *negative* lookahead and lookbehind.

RMcNeely replied:
I see the negative look ahead, I'm wondering what the reason to match the test lines `"```<span>should match</span>"` and `"<span>should match</span>```"` are. Those two seem like they symbolize the start of a multiline code block so I'm wondering why they are also included to be replaced instead of being left alone.

allen-ciq replied:
The current approach doesn't allow for multi-line code blocks.  The previous approach didn't respect code blocks at all; fixing that revealed a single case in our code where a `<span>` was embedded in a single-line code block (and it wasn't stripped/replaced because it was the second tag in the comment).  If we want to support embedded `<span>` tags in multi-line code blocks then a more comprehensive change may be needed.

RMcNeely reviewed (commented):
Can we turn these matches in the Regex101 link into unit tests so codify our expectations for the comment replacement?

allen-ciq committed:
39232 added unit tests

Added unit tests for replacement regex.  Also updated regex to allow
multiple `<span>` tags in the same line.
